homeassistant/components/homee/__init__.py

```diff
@@ -0,0 +1,274 @@
+    device_registry = dr.async_get(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        connections={
+            (dr.CONNECTION_NETWORK_MAC, dr.format_mac(homee.settings.mac_address))
+        },
+        identifiers={(DOMAIN, homee.settings.uid)},
+        manufacturer="homee",
+        name=homee.settings.homee_name,
+        model="homee",
+        sw_version=homee.settings.version,
+        hw_version="TBD",
+    )
+
+    # Forward entry setup to the platforms
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    entry.async_on_unload(entry.add_update_listener(async_update_entry))
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: HomeeConfigEntry) -> bool:
+    """Unload a homee config entry."""
+    # Unload platforms
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        # Get Homee object and remove it from data
+        homee: Homee = entry.runtime_data.homee
+
+        # Schedule homee disconnect
+        homee.disconnect()
+
+    return unload_ok
+
+
+async def async_update_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Reload homee integration after config change."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def async_remove_config_entry_device(
+    hass: HomeAssistant, config_entry: HomeeConfigEntry, device_entry: dr.DeviceEntry
+) -> bool:
+    """Remove a config entry from a device."""
+    homee = config_entry.runtime_data.homee
+    model = NodeProfile[device_entry.model].value
+    for node in homee.nodes:
+        # 'identifiers' is a set of tuples, so we need to check for the tuple.
+        if ("homee", node.id) in device_entry.identifiers and node.profile == model:
+            # If Node is still present in Homee, don't delete.
+            return False
+
+    return True
+
+
+class HomeeNodeEntity:
+    """Representation of a Node in Homee."""
+
+    _unrecorded_attributes = frozenset({ATTR_HOMEE_DATA})
+
+    def __init__(
+        self, node: HomeeNode, entity: Entity, entry: HomeeConfigEntry
+    ) -> None:
+        """Initialize the wrapper using a HomeeNode and target entity."""
+        self._node = node
+        self._entity = entity
+        self._clear_node_listener = None
+        self._attr_unique_id = node.id
+        self._entry = entry
+
+        self._homee_data = {
+            "id": node.id,
+            "name": node.name,
+            "profile": node.profile,
+            "attributes": [{"id": a.id, "type": a.type} for a in node.attributes],
+        }
+
+    async def async_added_to_hass(self) -> None:
+        """Add the homee binary sensor device to home assistant."""
+        self.register_listener()
+
+    async def async_will_remove_from_hass(self):
+        """Cleanup the entity."""
+        self.clear_listener()
+
+    @property
+    def device_info(self):
```

Comment:
This can move to `self._attr_device_info`

Reply:
And put the code in a seperate function?

Reply:
Nope, can be set in the constructor